jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererCodeCoverageTrackerTest.kt

```diff
@@ -207,6 +228,26 @@ class CodeWhispererCodeCoverageTrackerTest {
         assertThat(pythonTracker.totalTokensSize).isEqualTo(pythonTestLeftContext.length - 3)
     }
 
+    @Test
+    fun `test tracker documentChanged - will not increment tokens on blank string of length greater than 1`() {
+        val pythonTracker = TestCodePercentageTracker(
+            TOTAL_SECONDS_IN_MINUTE,
+            CodewhispererLanguage.Python,
+            codeCoverageTokens = mutableMapOf(fixture.editor.document to CodeCoverageTokens(pythonTestLeftContext.length, 0))
```

Comment:
why mutable

Reply:
@rli because we use mutable map in the tracker https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt#L38

Reply:
since we have to add the unseen file into this map and start counting tokens users are typing, and we reset every 5 mins